DigitalLearningSolutions.Data/DataServices/UserDataService/UserCentreDetailsDataService.cs

```diff
@@ -102,5 +110,23 @@ ELSE 0
                 transaction
             );
         }
+
+        public string? GetCentreEmail(int userId, int centreId)
+        {
+            var emails = connection.Query<string?>(
+                @"SELECT Email
+                    FROM UserCentreDetails
+                    WHERE UserID = @userId AND CentreID = @centreId",
+                new { userId, centreId}
+            ).ToList();
+            if (emails.Count > 1)
+            {
+                throw new MultipleUserAccountsFoundException(
+                    $"User with id {userId} has several emails at centre with id {centreId}"
+                );
+            }

```

Comment:
I don't think this exception is possible, there is a unique constraint on CentreID & Email (where not null), and one on CentreId & UserID, so each user can only have a single record per centre.